### ISSUE TITLE
error while installing dependencies for fbcunn

### ISSUE THREAD

**Issue by PraveerSINGH** [facebook/fbcunn#41]
inflating: gtest-1.7.0/README                                                                                                                                                                      [0/1300]
  inflating: gtest-1.7.0/configure.ac  
  inflating: gtest-1.7.0/aclocal.m4  
  inflating: gtest-1.7.0/Makefile.am  
  inflating: gtest-1.7.0/Makefile.in  
  inflating: gtest-1.7.0/configure   
  inflating: gtest-1.7.0/CHANGES     
  inflating: gtest-1.7.0/CONTRIBUTORS  
  inflating: gtest-1.7.0/LICENSE     
  inflating: gtest-1.7.0/CMakeLists.txt  
-- The C compiler identification is GNU 4.8.2
-- The CXX compiler identification is GNU 4.8.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Found Folly: /usr/local/include  
-- Found Glog: /usr/include  
-- Found Torch7 in /home/praveer/torch/install
CMake Error at /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
  REQUIRED_ARGS (missing: THRIFT_INCLUDE_DIR THRIFT_LIBRARY
  THRIFT_CPP2_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindThrift.cmake:79 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:27 (FIND_PACKAGE)

-- Configuring incomplete, errors occurred!
See also "/tmp/fblualib-build.q6E3tK/thpp/thpp/build/CMakeFiles/CMakeOutput.log".
Hey I receive the following error while installing the dependencies for fbcunn.
I have done all possible stuff, including installing manually folly (both using the github version and through folly) but all in vain. Any help would be highly appreciated :)

Installing FBLuaLib

-- The C compiler identification is GNU 4.8.2
-- The CXX compiler identification is GNU 4.8.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Found Folly: /usr/local/include  
-- Found Glog: /usr/include  
-- Found Torch7 in /home/praveer/torch/install
-- REQUIRED_ARGS (missing:  THPP_INCLUDE_DIR THPP_LIBRARIES) 
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
THPP_INCLUDE_DIR
   used as include directory in directory /tmp/fblualib-build.q6E3tK/fblualib/fblualib
THPP_LIBRARY
    linked by target "fblualib" in directory /tmp/fblualib-build.q6E3tK/fblualib/fblualib

-- Configuring incomplete, errors occurred!
See also "/tmp/fblualib-build.q6E3tK/fblualib/fblualib/build/CMakeFiles/CMakeOutput.log".

  inflating: gtest-1.7.0/README                                                                                                                                                                      [0/1300]
  inflating: gtest-1.7.0/configure.ac  
  inflating: gtest-1.7.0/aclocal.m4  
  inflating: gtest-1.7.0/Makefile.am  
  inflating: gtest-1.7.0/Makefile.in  
  inflating: gtest-1.7.0/configure   
  inflating: gtest-1.7.0/CHANGES     
  inflating: gtest-1.7.0/CONTRIBUTORS  
  inflating: gtest-1.7.0/LICENSE     
  inflating: gtest-1.7.0/CMakeLists.txt  
-- The C compiler identification is GNU 4.8.2
-- The CXX compiler identification is GNU 4.8.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Found Folly: /usr/local/include  
-- Found Glog: /usr/include  
-- Found Torch7 in /home/praveer/torch/install
CMake Error at /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
  REQUIRED_ARGS (missing: THRIFT_INCLUDE_DIR THRIFT_LIBRARY
  THRIFT_CPP2_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindThrift.cmake:79 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:27 (FIND_PACKAGE)

-- Configuring incomplete, errors occurred!
See also "/tmp/fblualib-build.q6E3tK/thpp/thpp/build/CMakeFiles/CMakeOutput.log".

Installing FBLuaLib

-- The C compiler identification is GNU 4.8.2
-- The CXX compiler identification is GNU 4.8.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Found Folly: /usr/local/include  
-- Found Glog: /usr/include  
-- Found Torch7 in /home/praveer/torch/install
-- REQUIRED_ARGS (missing:  THPP_INCLUDE_DIR THPP_LIBRARIES) 
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
THPP_INCLUDE_DIR
   used as include directory in directory /tmp/fblualib-build.q6E3tK/fblualib/fblualib
THPP_LIBRARY
    linked by target "fblualib" in directory /tmp/fblualib-build.q6E3tK/fblualib/fblualib

-- Configuring incomplete, errors occurred!
See also "/tmp/fblualib-build.q6E3tK/fblualib/fblualib/build/CMakeFiles/CMakeOutput.log".

All done!

**Comment by soumith** [comment]
looks like thrift and thpp were not installed correctly. what happened when they were compiling...?

**Comment by PraveerSINGH** [comment]
Found no issues while compiling them !!!

**Comment by szagoruyko** [comment]
Praveer can you paste the full log on https://gist.github.com ?

**Comment by cycladessoa-tangel** [comment]
Folks, could this possible be an API mismatch issue? After the "-e" flag was added to the bash command in the install directive, the error on my end was a bit more obvious:

-------
concurrency/ThreadManager.cpp:55:43:   required from here
/usr/local/include/folly/io/async/Request.h:124:43: error: 'static std::shared_ptr<folly::RequestContext>& folly::RequestContext::getStaticContext()' is private
   static std::shared_ptr<RequestContext> &getStaticContext() {
                                           ^
In file included from ../../../thrift/lib/cpp/concurrency/ThreadManager.tcc:40:0,
                 from concurrency/ThreadManager.h:398,
                 from concurrency/ThreadManager.cpp:17:
../../../thrift/lib/cpp/concurrency/ThreadManager-impl.h:127:40: error: within this context
       RequestContext::getStaticContext();
                                        ^
In file included from ../../../thrift/lib/cpp/async/Request.h:19:0,
                 from ../../../thrift/lib/cpp/concurrency/ThreadManager.tcc:26,
                 from concurrency/ThreadManager.h:398,
                 from concurrency/ThreadManager.cpp:17:
/usr/local/include/folly/io/async/Request.h:124:43: error: 'static std::shared_ptr<folly::RequestContext>& folly::RequestContext::getStaticContext()' is private
   static std::shared_ptr<RequestContext> &getStaticContext() {
                                           ^
In file included from ../../../thrift/lib/cpp/concurrency/ThreadManager.tcc:40:0,
                 from concurrency/ThreadManager.h:398,
                 from concurrency/ThreadManager.cpp:17:
../../../thrift/lib/cpp/concurrency/ThreadManager-impl.h:127:40: error: within this context
       RequestContext::getStaticContext();
                                        ^
-------


Without the "-e" it keeps running, and ultimately exits with the same error above.

**Comment by cycladessoa-tangel** [comment]
Adding the checkout command in your install script did the trick....apparently folly has been updated since your thrift version...:

cd $dir/folly/folly
>>> git checkout v0.37.0
autoreconf -ivf

**Comment by soumith** [comment]
sorry about this. I just fixed the fblualib install script so that this error does not happen now.
I made sure tagged versions of folly and fbthrift are installed, rather than trunk. I tested the install process on a fresh ubuntu 14.04 machine.

You'll have to redo the install process starting from this step:
https://github.com/facebook/fbcunn/blob/master/INSTALL.md#install-folly-fbthrift-thpp-and-fblualib

If you are having installation errors after retrying the install once more (after my fresh fixes), 
I can post the binaries here for Ubuntu 14.04 (they will be specific to ubuntu 14.04).
